hammer/hammer.go

```diff
@@ -182,6 +190,26 @@ func (h *Hammer) Run(ctx context.Context) {
 				if newSize > size {
 					klog.V(1).Infof("Updated checkpoint from %d to %d", size, newSize)
 				}
+				now := time.Now()
+				totalLatency := time.Duration(0)
+				queueLatency := time.Duration(0)
+				numLeaves := 0
+				for {
+					l, ok := <-h.leafSampleChan
+					if !ok {
+						break
+					}
+					if l.queuedAt.After(now) {
+						// We've caught up with everything which happened in this "epoch"
+						break
+					}
```

Comment:
This drops the write stat that was pulled into `l`. Are we happy with that? It doesn't seem like the end of the world, but it probably deserves a comment at least.